### PR TITLE
Rename and add test cases related to L7 endpoint calculation

### PR DESCRIPTION
**Pull request by sawsa307**
* Consolidate L7 endpoint calculation into TestValidateEndpointFields where it tests for both calculations

/assign @swetharepakula

### CODE REVIEW

File: pkg/neg/syncers/utils_test.go [hunk 1]
```diff
@@ -2138,19 +1771,79 @@ func TestDegradedModeValidateEndpointInfo(t *testing.T) {
 		negtypes.NetworkEndpoint{IP: "10.100.1.2", Node: instance1, Port: "80"}: types.NamespacedName{Namespace: testServiceNamespace, Name: "pod2"},
 	}
 
+	// endpointMapExcluded and podMapExcluded only includes valid endpoints.
+	// In normal mode, we exclude the specific endpoint for case where an endpoint's pod is invalid.
```
Comment (swetharepakula) [inline pkg/neg/syncers/utils_test.go:1775]: what do we mean by invalid in this case?

Comment (sawsa307) [reply pkg/neg/syncers/utils_test.go]: Just expanded comments to cover pod invalid situations.

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Comment by sawsa307** [comment]
Just resolved conflict with #2534.

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**